sympy/solvers/ode.py

```diff
@@ -1452,6 +1460,40 @@ def _test_term(coeff, order):
     else:
         return tuple(retlist)
 
+def separate_vars(f, symbols=[]):
+
+    from sympy import ordered
+
+    if symbols:
+        if not all((t.is_Atom for t in symbols)):
```

Comment:
why are you adding this restriction? Couldn't this separate an expression containing `f(x)` and `g(x)`?

Reply:
You could use `recast_to_symbols` (or similarly named function in solve, I believe).

Reply:
I am calling` _separatevars_dict` so it will raise the same value error. So I want to handle the error in the same function.

Reply:
> You could use `recast_to_symbols` (or similarly named function in solve, I believe).

I am looking to this.

Reply:
```
In [27]: eq=cos(x-y)+cos(x+y)                                                                 

In [28]: syms=[2*x,y]                                                                         

In [29]: recast_to_symbols([eq],syms)                                                         
Out[29]: ([cos(x - y) + cos(x + y)], [X₀, y], {X₀: 2⋅x})

```
As you can see in complex expression it will not work.

Reply:
> dsolve(Eq(f(x).diff(x),(cos(x+f(x))+cos(x-f(x)))),f(x))

notice the `a` bleeding through in the result...that's what I meant by the [comment](https://github.com/sympy/sympy/pull/16279#discussion_r266215345)